src/neovim/nvimUtil.ts

```diff
@@ -0,0 +1,65 @@
+"use strict";
+
+import * as vscode from "vscode";
+import { VimState } from "../mode/modeHandler";
+import { Position } from './../common/motion/position';
+import { TextEditor } from "../textEditor";
+import { Configuration } from '../configuration/configuration';
+
+export class Neovim {
+  // Data flows from VS to Vim
+  static async syncVSToVim(vimState: VimState) {
+    const nvim = vimState.nvim;
+    const buf = await nvim.getCurrentBuf();
+    await buf.setLines(0, -1, true, TextEditor.getText().split('\n'));
+
+    await nvim.callFunction("setpos", [".", [0, vimState.cursorPosition.line + 1, vimState.cursorPosition.character, false]]);
+    await nvim.callFunction("setpos", ["'>", [0, vimState.cursorPosition.line + 1, vimState.cursorPosition.character, false]]);
+    await nvim.callFunction("setpos", ["'<", [0, vimState.cursorStartPosition.line + 1, vimState.cursorStartPosition.character, false]]);
+    for (const mark of vimState.historyTracker.getMarks()){
+      await nvim.callFunction("setpos", [`'${mark.name}`, [0, mark.position.line + 1, mark.position.character, false]]);
+    }
+  }
+
+  // Data flows from Vim to VS
+  static async syncVimToVs(vimState: VimState) {
+    const nvim = vimState.nvim;
+    const buf = await nvim.getCurrentBuf();
+
+    await TextEditor.replace(
+    new vscode.Range(0, 0, TextEditor.getLineCount() - 1,
+    TextEditor.getLineMaxColumn(TextEditor.getLineCount() - 1)),
+    (await buf.getLines(0, -1, false)).join('\n')
+    );
+
+    let [row, character]  = (await nvim.callFunction("getpos", ["."]) as Array<number>).slice(1, 3);
+    vimState.editor.selection = new vscode.Selection(new Position(row - 1, character), new Position(row - 1, character));
+
+    if (Configuration.expandtab) {
+      await vscode.commands.executeCommand("editor.action.indentationToSpaces");
+    }
+  }
+
+  static async command(vimState: VimState, command: string) {
+    const nvim = vimState.nvim;
+    await this.syncVSToVim(vimState);
+    command = ":" + command + "\n";
```

Comment:
Yes it does. We're already handling it here: https://github.com/Chillee/Vim/blob/24b1a348fa286d8825275fb7ec47b70b8d9c3ee3/src/actions/commands/actions.ts#L1449

We need to handle it before we pass it to neovim so that the user can see it in their command line.